crates/nu-command/src/filters/uniq.rs

```diff
@@ -201,6 +213,41 @@ fn clone_to_lowercase(value: &Value) -> Value {
     }
 }
 
+fn sort_attributes(val: Value) -> Value {
+    match val {
+        Value::Record { cols, vals, span } => {
+            let sorted = cols
+                .into_iter()
+                .zip(vals)
+                .sorted_by(|a, b| a.0.cmp(&b.0))
+                .collect_vec();
+
+            let sorted_cols = sorted.clone().into_iter().map(|a| a.0).collect_vec();
+            let sorted_vals = sorted
+                .into_iter()
+                .map(|a| sort_attributes(a.1))
+                .collect_vec();
+
+            Value::Record {
+                cols: sorted_cols,
+                vals: sorted_vals,
+                span,
+            }
+        }
+        Value::List { vals, span } => Value::List {
+            vals: vals.into_iter().map(sort_attributes).collect_vec(),
+            span,
+        },
+        other => other,
+    }
+}
+
+fn generate_key(item: &ValueCounter) -> String {
+    let value = sort_attributes(item.val_to_compare.clone()); //otherwise, keys could be different for Records
+    value_to_string(&value, Span::unknown())
+        .unwrap_or_else(|_| panic!("Could not convert to string: {:?}", value))
```

Comment:
I think panicking here is a bad idea - `uniq` should not be able to crash Nu. Maybe it would make more sense for `generate_key` to return a `Result<String, ShellError>`?

Reply:
that's right. I will change that